### PR TITLE
feat(subvec): add sub vector method including remainder

### PR DESCRIPTION
**Pull request by Maxpicca-Li**
add sub vector method including remainder, which can be used to support multi-port output for queue/buffer.

### CODE REVIEW

File: src/main/scala/utility/SubVec.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    require(in.size % mod == 0)
+    require(mod > rem)
+    (0 until in.size / mod).map(i => {in(mod * i + rem)})
+  }
+  def getRems[T <: Data](in: Seq[T], mod: Int): Seq[Seq[T]] = {
+    (0 until mod).map { rem => getRem(in, mod, rem) }
+  }
+
+  // RemWithExpand: SubSeqWithExpand = (Seq % Modulus expanded)(Remainder)
+  def getRemWithExpand(in: Seq[Bool], mod: Int, rem: Int): Seq[Bool] = {
```
Comment (cz4e) [inline src/main/scala/utility/SubVec.scala]: Can it be other types?

File: src/main/scala/utility/SubVec.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    require(mod > rem)
+    val subSize = in.size / mod + 1
+    (0 until subSize).map {i =>
+      val idx = mod * i + rem
+      if (idx < in.size) in(idx) else false.B
+    }
+  }
+  def getRemWithExpands(in: Seq[Bool], mod: Int): Seq[Seq[Bool]] = {
```
Comment (cz4e) [inline src/main/scala/utility/SubVec.scala]: Can you change the name of the function？For example, getRem can be called do_unzip or do_extract, and getMaskRem can be called do_zip or do_insert.

**Review by cz4e — APPROVED** [review]

File: src/main/scala/utility/SubVec.scala (hunk 1 above)
Comment (Maxpicca-Li) [reply src/main/scala/utility/SubVec.scala]: Yes. Now it changed to T

File: src/main/scala/utility/SubVec.scala (hunk 2 above)
Comment (Maxpicca-Li) [reply src/main/scala/utility/SubVec.scala]: The names 'zip' or 'extract' feel unintuitive, so the original names are retained here.

**Pull request merged**